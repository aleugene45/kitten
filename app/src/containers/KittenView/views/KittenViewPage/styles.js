import { StyleSheet } from 'react-native';
import themes from "../../../../../themes";

export default StyleSheet.create({
  wrapper: {
    flex: 1,
  },
  image: {
    flex: 4,
  },
  infoContainer: {
    flex: 6,
    margin: 15,
    flexDirection: 'column'
  },
  title: {
    fontSize: themes.fontSize.large,
    color: 'black',
    fontWeight: 'bold'
  },
  info: {
    fontSize: themes.fontSize.default,
    color: 'black',
  }
});
