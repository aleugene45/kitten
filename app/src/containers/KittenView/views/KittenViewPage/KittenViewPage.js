import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {
  View,
  Text,
  ActivityIndicator
} from 'react-native';
import Image from 'react-native-image-progress'
import styles from './styles'

const lorem = 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.';

class KittenViewPage extends PureComponent {
  render() {
    return <View style={styles.wrapper}>
      <Image
        style={styles.image}
        indicator={() => <ActivityIndicator color={'black'} size={40}/>}
        source={{uri: this.props.kitten.imageUrl}}
        resizeMode={'cover'}
      />
      <View style={styles.infoContainer}>
        <Text style={styles.title}>{this.props.kitten.label}</Text>
        <Text style={styles.info}>{lorem}</Text>
      </View>
    </View>
  }
}

KittenViewPage.propTypes = {
  kitten: PropTypes.object.isRequired,
};

export default KittenViewPage;
