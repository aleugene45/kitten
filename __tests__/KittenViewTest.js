/**
 * @format
 */

import 'react-native';
import React from 'react';
import uuid from 'react-native-uuid';
import KittenViewPage from '../app/src/containers/KittenView/views/KittenViewPage/KittenViewPage';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('KittenViewPage renders correctly', () => {
  const mockItem = {
    imageUrl: `http://placekitten.com/200/300?image=1`,
    label: 'Bateman',
    id: uuid.v4()
  };

  const navigation = { navigate: jest.fn() };

  renderer.create(<KittenViewPage kitten={mockItem} navigation={navigation}/>);
});

it('KittenViewPage match snapshot', () => {
  const mockItem = {
    imageUrl: `http://placekitten.com/200/300?image=1`,
    label: 'Bateman',
    id: uuid.v4()
  };

  const navigation = { navigate: jest.fn() };

  const component = renderer.create(<KittenViewPage kitten={mockItem} navigation={navigation}/>).toJSON();
  expect(component).toMatchSnapshot();
});
