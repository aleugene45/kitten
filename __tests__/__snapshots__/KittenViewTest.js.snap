// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KittenViewPage match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 4,
      }
    }
  >
    <Image
      onError={[Function]}
      onLoad={[Function]}
      onLoadEnd={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      resizeMode="cover"
      source={
        Object {
          "uri": "http://placekitten.com/200/300?image=1",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 6,
        "flexDirection": "column",
        "margin": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Bateman
    </Text>
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 16,
        }
      }
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
    </Text>
  </View>
</View>
`;
